### PR TITLE
Add SetStage() calls to ICC encounters for phase tracking

### PR DESCRIPTION
**Pull request by Fawenah**
Adds a SetStage(phase) call to all the ICC encounters.

It is currently untested as no PTR available
Changes should be minimal enough to not be breaking.

### CODE REVIEW

File: Citadel/Festergut.lua [hunk 1]
```diff
@@ -53,6 +54,7 @@ function mod:OnBossEnable()
 end
 
 function mod:OnEngage()
+	self:SetStage(1)
```
Comment (Justw8) [inline Citadel/Festergut.lua:57]: You don't have to SetStage on WarmUp/OnEngage if the encounter never changes stages :)!

File: Citadel/Rotface.lua [hunk 2]
```diff
@@ -51,6 +52,7 @@ function mod:OnBossEnable()
 end
 
 function mod:OnEngage()
+	self:SetStage(1)
```
Comment (Justw8) [inline Citadel/Rotface.lua:55]: Just as the others, don't need to setstage in OnEngage.

File: Citadel/Saurfang.lua [hunk 3]
```diff
@@ -118,6 +120,7 @@ function mod:Mark(args)
 end
 
 function mod:Frenzy(args)
+	self:SetStage(2)
```
Comment (Justw8) [inline Citadel/Saurfang.lua:123]: Do you really want to change stage for frenzy? do the timers change? Usually a SetStage really means the fight changes - not just when a boss does more damage.

Comment (Justw8) [reply Citadel/Saurfang.lua:123]: And keep in mind, players can just do a Boss > X health check for these things (or active buff)

File: Citadel/Valithria.lua [hunk 4]
```diff
@@ -86,6 +87,7 @@ do
 		end
 	end
 	function mod:OnEngage()
+		self:SetStage(1)
```
Comment (Justw8) [inline Citadel/Valithria.lua:90]: As others, no need in OnEnrage

File: Citadel/Saurfang.lua (hunk 3 above)
Comment (Fawenah) [reply Citadel/Saurfang.lua:123]: Probably not. I just tried to think of what "could" be considered a new phase. Will remove.

**Comment by Fawenah** [comment]
> Left some comments, sorry for them being all in their own post <_<!

Makes it easier to track all the points they refer to though

File: Citadel/Putricide.lua [hunk 5]
```diff
@@ -128,6 +130,11 @@ do
 		self:Bar("phase", 18, L["phase_bar"], "achievement_boss_profputricide")
 		self:ScheduleTimer(nextPhase, 3)
 		stopOldStuff()
+		if not first then
+			self:SetStage(2)
```
Comment (Justw8) [inline Citadel/Putricide.lua:134]: Why not just place these with the messages of stage 2 and stage 3? One would expect these to be when we announce them in the messages.

**Comment by Justw8** [comment]
One last note after looking into them more precisely :)

**Comment by Justw8** [comment]
https://github.com/BigWigsMods/BigWigs_WrathOfTheLichKing/actions/runs/6446802556/job/17504895818?pr=26

some errors on the luacheck left now ^^

**Comment by Justw8** [comment]
Thanks! enjoy ICC in a few days :)

**Pull request merged**